internal/tracinginterceptor/interceptor.go

```diff
@@ -90,3 +186,48 @@ func (m *Interceptor) CallStream(ctx context.Context, req *transport.StreamReque
 	// TODO: implement
 	panic("implement me")
 }
+
+func updateSpanWithError(span opentracing.Span, err error) error {
+	if err == nil {
+		return err
+	}
+
+	ext.Error.Set(span, true)
+	if yarpcerrors.IsStatus(err) {
+		status := yarpcerrors.FromError(err)
+		errCode := status.Code()
+		span.SetTag("rpc.yarpc.status_code", errCode.String())
+		span.SetTag("error.type", errCode.String())
+		return err
+	}
+
+	span.SetTag("error.type", "unknown_internal_yarpc")
+	return err
+}
+
+func updateSpanWithOutboundError(span opentracing.Span, res *transport.Response, err error) error {
```

Comment:
Is it indeed for "outbound"? We probably may call it on inbound side, but when we already have response?

Reply:
Thanks, renamed to updateSpanWithResponseError